src/gpu-compute/fetch_unit.cc

```diff
@@ -471,6 +493,18 @@ FetchUnit::FetchBufDesc::reserveBuf(Addr vaddr)
 void
 FetchUnit::FetchBufDesc::fetchDone(Addr vaddr)
 {
+    // If the return vaddr is 0, then it belongs to an SQC invalidation
+    // request. This request calls incLGKMInstsIssued() function in its
+    // execution path. Since there is no valid memory return response
+    // associated with this instruction, decLGKMInstsIssued() is not
+    // executed. Do this here to decrement the counter and invalidate
+    // all buffers
+    if (vaddr == 0) {
```

Comment:
This method seems to be called in one place only which has access to the packet.  I think it would be safer to change the method to pass the packet, then here it could check if it is a MemSyncResp (or whatever SQC invalidate response is)

Reply:
I updated the function to use the packet command instead. Thanks for the suggestion!

Reply:
good call, @abmerop !